lib/rom/repository/command_compiler.rb

```diff
@@ -208,7 +208,10 @@ def finalize_command_class(klass, relation)
 
         klass.extend_for_relation(relation) if klass.restrictable
 
-        plugins.each { |plugin| klass.use(plugin) }
+        plugins.each do |(plugin, block)|
+          klass.use(plugin) 
+          klass.instance_eval(&block) if block
```

Comment:
@flash-gordon OK, thanks to new features in commands I fixed the issue with caching, see here: https://github.com/rom-rb/rom-repository/pull/43/commits/f5f62477c8ed80b2ad1a1152b74360784f9b5f52